File: frontend/src/types/invoice.ts
export type InvoiceStatus = 'draft' | 'sent' | 'paid' | 'overdue' | 'cancelled'
export type PaymentStatus = 'paid' | 'partial' | 'unpaid'

export interface InvoiceLedgerItem {
  invoice_id: string
  client_name: string
  client_country: string
  issue_date: string
  due_date: string
  amount: number
  currency: string
  invoice_status: InvoiceStatus
  total_paid: number
  payment_count: number
  balance_due: number
  payment_status: PaymentStatus
}

export interface InvoiceLedgerResponse {
  data: InvoiceLedgerItem[]
  total: number
}

File: backend/app/schemas/invoice.py
from datetime import date

from pydantic import BaseModel


class InvoiceLedgerItem(BaseModel):
    invoice_id: str
    client_name: str
    client_country: str
    issue_date: date
    due_date: date
    amount: float
    currency: str
    invoice_status: str
    total_paid: float
    payment_count: int
    balance_due: float
    payment_status: str


class InvoiceLedgerResponse(BaseModel):
    data: list[InvoiceLedgerItem]
    total: int
